main.tf

```diff
@@ -82,29 +82,79 @@ resource "aws_backup_selection" "main" {
   }
 }
 
+#######
+# SNS Backup Notifications
+#######
+resource "aws_sns_topic" "main" {
+  count = var.sns_topic_arn != null ? 0 : 1
+
+  name              = var.vault_name != null ? "${aws_backup_vault.main[0].name}-events" : "backup-vault-events"
+  kms_master_key_id = var.vault_sns_kms_key_arn
+}
+
+data "aws_iam_policy_document" "sns_policy" {
+  count = var.enable_sns_notifications ? 1 : 0
+
+  statement {
+    sid = "AllowSNSPublish"
+
+    actions = [
+      "SNS:Publish",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["backup.amazonaws.com"]
+    }
+
+    resources = var.sns_topic_arn != null ? [var.sns_topic_arn] : [aws_sns_topic.main[0].arn]
+  }
+}
+
+resource "aws_sns_topic_policy" "main" {
+  count = var.enable_sns_notifications ? 1 : 0
+
+  arn    = var.sns_topic_arn != null ? var.sns_topic_arn : aws_sns_topic.main[0].arn
+  policy = data.aws_iam_policy_document.sns_policy[0].json
+}
+
+resource "aws_backup_vault_notifications" "main" {
+  count = var.enable_sns_notifications ? 1 : 0
+
+  backup_vault_name   = var.vault_name != null ? aws_backup_vault.main[0].name : "Default"
+  sns_topic_arn       = var.sns_topic_arn != null ? var.sns_topic_arn : aws_sns_topic.main[0].arn
+  backup_vault_events = var.backup_vault_events
+}
+
 #####
 # IAM Role
 #####
 resource "aws_iam_role" "main" {
   name               = var.iam_role_name != null ? var.iam_role_name : "aws-backup-plan-${var.plan_name}-role"
-  assume_role_policy = <<POLICY
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Action": ["sts:AssumeRole"],
-      "Effect": "allow",
-      "Principal": {
-        "Service": ["backup.amazonaws.com"]
-      }
-    }
-  ]
-}
-POLICY
+  assume_role_policy = data.aws_iam_policy_document.main.json
 
   tags = var.tags
 }
 
+data "aws_iam_policy_document" "main" {
+  statement {
+    sid = "AllowBackupService"
+
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["backup.amazonaws.com"]
```

Comment:
Could you please check whether service identifier is the same for China region?

Reply:
I just had a quick look - you're right it's different:  `backup.cn-north-1.amazonaws.com.cn` from [this doc about endpoints](https://docs.amazonaws.cn/en_us/aws/latest/userguide/endpoints-Beijing.html)

I'll be sure to update this for adoption of China region.

Reply:
Turns out we can still keep this at `backup.amazonaws.com`

Reply:
👍